main.go

```diff
@@ -7,35 +7,98 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
+
+	"github.com/tcnksm/go-gitconfig"
 )
 
 //go:generate go-bindata _templates/...
 
+func usage() {
+	fmt.Fprintf(os.Stderr, `Usage of %s:
+	%s new <project name>
+	%s gen -d <model directory -o <output directory>`,
+	os.Args[0], os.Args[0], os.Args[0])
+	os.Exit(1)
+}
+
 func main() {
-	var (
-		modelDir string
-		outDir   string
-	)
-
-	flag.Usage = func() {
-		fmt.Fprintf(os.Stderr, `Usage of %s:
-   %s -d <model directory> -o <output directory>
-
-Options:
-`, os.Args[0], os.Args[0])
-		flag.PrintDefaults()
+
+	if len(os.Args) < 2 {
+		usage()
+	}
+
+	cmd := os.Args[1]
+
+	switch cmd {
+	case "gen":
+		var (
+			modelDir string
+			outDir   string
+		)
+
+		flag := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+		flag.Usage = func() {
+			fmt.Fprintf(os.Stderr, `Usage of %s:
+	   %s gen -d <model directory> -o <output directory>
+
+	Options:
+	`, os.Args[0], os.Args[0])
+			flag.PrintDefaults()
+		}
+
+		flag.StringVar(&modelDir, "d", "", "Model directory")
+		flag.StringVar(&outDir, "o", "", "Output directory")
+
+		flag.Parse(os.Args[:2])
+
+		if modelDir == "" || outDir == "" {
+			flag.Usage()
+			os.Exit(1)
+		}
+		cmdGen(modelDir, outDir)
+
+	case "new":
+		if len(os.Args) < 3 {
+			fmt.Fprintf(os.Stderr, `Usage of %s:
+	   %s gen new <model name>`, os.Args[0], os.Args[0])
+			os.Exit(1)
+		}
+
+		name := os.Args[2]
+		cmdNew(name)
+
+	default:
+		usage()
+	}
+
+}
+
+func cmdNew(name string) {
+	vcs := "github.com"
+	username, err := gitconfig.Username()
```

Comment:
It's better to fetch _GitHub_ username before fetching _git_ username.

Some developers set their realname as their git username, so that directory path will become invalid.

In my case (git name: `Daisuke Fujita`):

```bash
$ ./api-server-generator new hoge
mkdir /Users/dtan4/src/github.com/Daisuke Fujita/hoge: no such file or directory
```

## REF
- https://github.com/tcnksm/gcli/blob/ea36f4d2935a7519a77b4027bbafb425f37236a2/command/new.go#L73-L87
  - Example how to use `gitconfig` in gcli